app/controllers/feedback_controller.rb

```diff
@@ -29,30 +29,31 @@ def show
   private
 
   def provider_email
-    if params[:signed_out_provider_id].present?
-      signed_out_provider_email
-    else
-      signed_in_provider_email
-    end
+    # citizen side won't have access to current_provider but can get provider_email via current_application_id
+    provider&.email || current_provider&.email
   end
 
-  def signed_out_provider_email
-    provider_id = params[:signed_out_provider_id]
-    Provider.find(provider_id).email
+  def provider
+    LegalAidApplication.find_by_id(application_id)&.provider
   end
 
-  def signed_in_provider_email
-    provider_struct = session['warden.user.provider.key']
-    return nil unless provider_struct
+  def application_id # rubocop:disable Metrics/AbcSize
+    return params['application_id'] if params['application_id']
+    return session['current_application_id'] if source == :citizen
+
+    application_id_from_page_history || nil
+  end
 
-    provider_id = provider_struct.first&.first
-    return nil unless provider_id
+  def application_id_from_page_history
+    previous_page = session['page_history'][-2]
 
-    Provider.find(provider_id).email
+    previous_page&.split('/')&.each_with_index do |section, i|
+      return session['page_history'][-2].split('/')[i + 1] if ['applications'].include?(section)
```

Comment:
This line will also need to change because of the new way of handling page histories.